### ISSUE TITLE
request: send to repl

### ISSUE THREAD

**Issue by fdschmidt93** [mfussenegger/nvim-dap#504]
### Problem Statement

tl;dr: ability to send lines from the buffer to the dap-repl.

My idea would be (probably a personal) plugin that reimplements https://github.com/jpalardy/vim-slime with `nvim_chan_send` (I have some code for Python, but it was never worthwhile finishing as vim-slime works too well) with the ability to also send to the prompt (or maybe kernel directly?) as well.

### Ideas or possible solutions

I've got an initial solution working by simply exposing `execute` in `repl` (`M.execute = execute`). I've played around with it as per the video, which also shows something that would have to be fixed I suppose when printing to the prompt buffer :)

https://user-images.githubusercontent.com/39233597/161929565-963eec45-9ae6-450c-be77-47efcdb38a90.mp4

I'm happy to eventually make a PR for this, though as I'm unfamiliar with DAP, adapters etc., assuming this feature would be desirable, I initially wanted to get some thoughts/considerations (anything I should know?) on how to provide the functionality to programmatically send lines to the prompt buffer or kernel directly (whatever is more suitable, though exposing `execute` in some form seems like the way to go at first glance).

e: As an alternative, yanking in buffer and then `<C-r>0` also works really well already.

**Comment by mfussenegger** [comment]
You can append lines to the repl using `require('dap.repl').append('text')`. `execute` is not the same - it would also evaluate the text (first checking internal commands, and if none match it uses the debug adapter `evaluate` request). This is not something I want to expose.

If you do want to fire off evaluate requests you can interact with the session directly:

```lua
local session = dap.session()
if session then
  session:evaluate('expression to evaluate', function(err, resp)
  end)
end
```

**Comment by fdschmidt93** [comment]
A combination sending visual selection via `append` and `session:evaluate` works like a charm, thank you! :)
